pyfr/integrators/dual/phys/steppers.py

```diff
@@ -1,25 +1,127 @@
 # -*- coding: utf-8 -*-
 
+import math
+
 from pyfr.integrators.dual.phys.base import BaseDualIntegrator
 
 
 class BaseDualStepper(BaseDualIntegrator):
-    pass
 
+    def _finalize_step(self):
+        pass
+
+
+class BaseBDFStepper(BaseDualStepper):
+    nstages = 1
+    stage_nregs = 0
+    c = [1]
+
+    @property
+    def stepper_nregs(self):
+        return len(self.stepper_static_coeffs) - 1
+
+    def _finalize_step(self):
+        psnregs = self.pseudointegrator.pseudo_stepper_nregs
+        snregs = self.pseudointegrator.stepper_nregs
+
+        # Rotate the source registers to the right by one
+        self.pseudointegrator._regidx[psnregs:psnregs + snregs] = (
+            self.pseudointegrator._stepper_regidx[-1:]
+            + self.pseudointegrator._stepper_regidx[:-1]
+        )
+
+        # Copy the current soln into the first source register
+        self.pseudointegrator._add(
+            0, self.pseudointegrator._stepper_regidx[0],
+            1, self.pseudointegrator._idxcurr
+        )
 
-class DualBDF2Stepper(BaseDualStepper):
+    def stepper_coeffs(self, s, dt):
+        return [1] + [sc/dt for sc in self.stepper_static_coeffs]
+
+
+class DualBDF2Stepper(BaseBDFStepper):
     stepper_name = 'bdf2'
     stepper_order = 2
-    stepper_coeffs = [-1.5, 2.0, -0.5]
+    stepper_static_coeffs = [-1.5, 2.0, -0.5]
 
 
-class DualBDF3Stepper(BaseDualStepper):
+class DualBDF3Stepper(BaseBDFStepper):
     stepper_name = 'bdf3'
     stepper_order = 3
-    stepper_coeffs = [-11.0/6.0, 3.0, -1.5, 1.0/3.0]
+    stepper_static_coeffs = [-11.0/6.0, 3.0, -1.5, 1.0/3.0]
 
 
-class DualBackwardEulerStepper(BaseDualStepper):
+class DualBackwardEulerStepper(BaseBDFStepper):
```

Comment:
I think the plan is to move away from BDF in the future. Do we want to change the Backwards Euler to be a child of BaseDIRKStepper? That way we can just chop out BDF later? a = [1], b = [1]